### ISSUE TITLE
access all line information for appointment

### ISSUE THREAD

**Issue by turkdoktoru** [DevExpress/devextreme-reactive#2839]
I use the appointment system with react. Editable appointment.
I only reach the edited area change.
How can I get all the line information of the appointment I have arranged?

I want to access the line information for "**changed**" and "**deleted**"

`

        commitChanges({ added, changed, deleted }) {
         let { appointmentsItems } = this.state;

         if (changed) {
           //const data = appointmentsItems.map(appointment => (
           //  changed[appointment.id] ? { ...appointment, ...changed[appointment.id] } : appointment));

           const data = appointmentsItems.map(appointment => ( changed[appointment.id] ? {
             ...appointment,
             ...changed[appointment]
           } : appointment));
           console.log(changed)
           console.log(data);
         }
       }

**Comment by AryamnovEugeniy** [comment]
Hello,

`changed` is an object containing `key` fields. Every key specifies the associated appointment's ID. `deleted` is the identifier of an appointment to be deleted (refer to [our documentation](https://devexpress.github.io/devextreme-reactive/react/scheduler/docs/reference/editing-state/#changeset) for more information). Both of these parameters use appointments' ids. So, for editing to work correctly, every [appointment](https://devexpress.github.io/devextreme-reactive/react/scheduler/docs/reference/scheduler/#appointmentmodel) must have a unique id. 

In other words, the solution is to add `id` field to your appointments and follow one of our [examples](https://devexpress.github.io/devextreme-reactive/react/scheduler/docs/guides/editing/).
